### DISCUSSION TITLE
Install with pip or brew? I have both

### DISCUSSION THREAD

**Discussion by BarbzYHOOL** [dooit-org/dooit#153] [Q&A] [answered]
Hello,

I have homebrew for a few days now, and pip since forever.
I wonder which one is better, what are the pros and cons for installing this tool?

Maybe using pip, it will install dependencies reusable by other python programs? no idea

**Comment by kraanzu** [comment]
Hey @BarbzYHOOL, I have no idea how brew packaging works
[This](https://github.com/Homebrew/homebrew-core/pull/104251/files#diff-61a19f272132820f0d59501a289804406874c7e1dfcc5efdc6927a7062658e3f) was the pull request for homebrew

And [this](https://github.com/Homebrew/homebrew-core/blob/5d24fbfc322ff57d0cc0cf489346b6e80cfeb975/Formula/d/dooit.rb) is the latest file which looks like they use exact versioning for dependencies which can cause some issues.

As for pip, yes other programs can reuse the dependencies but this again has versioning issues for example one of my old projects still uses the older tui interface which is not compatible with the version dooit uses

One thing you can do is directly get the [prepared binary from releases](https://github.com/kraanzu/dooit/releases/tag/v2.1.0)  or you can look into [pipx](https://pypa.github.io/pipx/) which installs the application in isolation so that other programs are not affected.

This takes up a little bit more space for example dooit binary for Mac OS takes up 10MB which is fairly small and should not be an issue :)

**Reply by BarbzYHOOL** [reply]
getting the binary is cool but to update, it gives more work, right?


so now you told me about pipx (which is described as similar to brew) and i'm even more confused, all these things seem like the same to me, except I had issues once with pip in the past

**Reply by kraanzu** [reply]
> getting the binary is cool but to update, it gives more work, right?

What do you mean by "more work", like downloading and then placing it in the right place?
If yes, then I Don't know about mac but for example, AUR lets you do that for arch linux ([see this](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=dooit))

Yes, there are multiple ways but IMO whichever provides the isolated environment is better :D

**Reply by BarbzYHOOL** [reply]
yes, downloading manually and adding it in the right place (kinda defeats the practicability of using package managers if we do this by hand lol)

well ok for isolated environment, but again, isn't the point of package managers to reuse libraries?

**Reply by kraanzu** [reply, marked as answer]
> isn't the point of package managers to reuse libraries

Well, you can say that but not everyone uses the same versions right?
For projects that are mature enough that the only changes will be mostly bug fixes/little improvements

But for example, dooit uses [textual](https://github.com/Textualize/textual) which is very much in the growing stage and changes pretty often if you look at their [releases](https://github.com/Textualize/textual/releases)

That's also the reason I had to prepare a binary for [AUR packages](https://aur.archlinux.org/) because the version of textual there was old and was not compatible with dooit's latest version

So in short, it depends on what you want to use :)
I hope it makes sense

**Reply by BarbzYHOOL** [reply]
yes it does, it helps me in choosing which pkg for Dooit
